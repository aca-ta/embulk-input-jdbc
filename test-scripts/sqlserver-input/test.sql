DROP TABLE EMBULK_INPUT;

CREATE TABLE EMBULK_INPUT (
    ID     DECIMAL(4, 0),
    NUM    DECIMAL(12,2),
    STR    CHAR(8),
    VARSTR VARCHAR(8),
    DT     DATE,
    DTTM   DATETIME,
    DTTM2  DATETIME2,
    SDTTM   SMALLDATETIME,
    PRIMARY KEY(ID)
);


INSERT INTO EMBULK_INPUT VALUES(
  1,
  123.4,
  'test1',
  'TEST1',
  '2015-04-24',
  '2015-04-24 01:02:03.123',
  '2015-04-24 01:02:03.1234567',
  '2015-04-24 01:02:03'
);

INSERT INTO EMBULK_INPUT VALUES(
  2,
  1234567890.12,
  'test9999',
  'TEST9999',
  '2015-12-31',
  '2015-12-31 23:59:59.993',
  '2015-12-31 23:59:59.9999999',
  '2015-12-31 23:59:59'
);

INSERT INTO EMBULK_INPUT VALUES(
  3,
  NULL,
  NULL,
  NULL,
  NULL,
  NULL,
  NULL,
  NULL
);
