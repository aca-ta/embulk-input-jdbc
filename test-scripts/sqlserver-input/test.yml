in:
  type: sqlserver
  driver_path: driver\sqljdbc41.jar
  application_name: embulk_test
  host: localhost
  port: 1433
  database: TESTDB
  user: TEST_USER
  password: XXXXXXXX
  table: EMBULK_INPUT
  select: "*"
  column_options:
    ID: {type: string}
    NUM: {type: string}
    DT: {type: string, timestamp_format: '%Y-%m-%d', timezone: "+0900"}
    DTTM: {type: string, timestamp_format: '%Y-%m-%d %H:%M:%S.%3N', timezone: "+0900"}
    DTTM2: {type: string, timestamp_format: '%Y-%m-%d %H:%M:%S.%7N', timezone: "+0900"}
    SDTTM: {type: string, timestamp_format: '%Y-%m-%d %H:%M:%S', timezone: "+0900"}
out:
  type: file
  path_prefix: data/test
  file_ext: csv
  formatter:
    type: csv
