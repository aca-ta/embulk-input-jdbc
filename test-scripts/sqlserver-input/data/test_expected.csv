ID,NUM,STR,VARSTR,DT,DTTM,DTTM2,SDTTM
1,123.40,test1   ,TEST1,2015-04-24,2015-04-24 01:02:03.123,2015-04-24 01:02:03.1234567,2015-04-24 01:02:00
2,1234567890.12,test9999,TEST9999,2015-12-31,2015-12-31 23:59:59.993,2015-12-31 23:59:59.9999999,2016-01-01 00:00:00
3,,,,,,,
